src/sentry/static/sentry/app/views/stream/organizationSavedSearchSelector.jsx

```diff
@@ -126,13 +140,48 @@ const SaveSearchButton = withApi(
         isSaving: false,
         query: props.query,
         name: '',
+        error: null,
       };
     }
 
     onSubmit = e => {
+      const {api, organization, onSave} = this.props;
+
       e.preventDefault();
 
-      // TODO: implement saving
+      this.setState({isSaving: true});
+
+      addLoadingMessage(t('Saving Changes'));
+
+      api.request(`/organizations/${organization.slug}/searches/`, {
```

Comment:
Should this use an actionCreator?

Reply:
I'd prefer not to use an action creator when we're not updating a store or any aspect of global state, I think we currently use this pattern more commonly

Reply:
I was more interested in separating the rendering logic and logic that does API interactions.

There are some tasks in the backlog to move saved-searches into a store, we could move this then.